Library/Homebrew/os/linux/elf.rb

```diff
@@ -89,13 +89,16 @@ def with_interpreter?
   def dynamic_elf?
```

Comment:
Nit: For cached functions like these, combining the memoize + access like this is a little more readable IMO:

```ruby
@dynamic_elf ||= begin
  if HOMEBREW_READELF_RB
    # ...
  elsif which "readelf"
    # ...
  else
    # ...
  end
end
```

Reply:
(You might be able to do it with `||= if ...` directly, I don't remember off the top of my head if that `begin` block is required)

Reply:
Thanks for the change!